cql/src/main/java/org/apache/cassandra/stargate/transport/internal/Event.java

```diff
@@ -106,31 +111,35 @@ private NodeEvent(Type type, InetSocketAddress node) {
 
     public final Change change;
 
-    private TopologyChange(Change change, InetSocketAddress node) {
-      super(Type.TOPOLOGY_CHANGE, node);
+    private TopologyChange(
+        Change change, InetSocketAddress node, Predicate<Map<String, String>> headerFilter) {
+      super(Type.TOPOLOGY_CHANGE, node, headerFilter);
       this.change = change;
     }
 
-    public static TopologyChange newNode(InetAddressAndPort address) {
+    public static TopologyChange newNode(
+        InetAddressAndPort address, Predicate<Map<String, String>> headerFilter) {
       return new TopologyChange(
-          Change.NEW_NODE, new InetSocketAddress(address.address, address.port));
+          Change.NEW_NODE, new InetSocketAddress(address.address, address.port), headerFilter);
     }
 
-    public static TopologyChange removedNode(InetAddressAndPort address) {
+    public static TopologyChange removedNode(
+        InetAddressAndPort address, Predicate<Map<String, String>> headerFilter) {
       return new TopologyChange(
-          Change.REMOVED_NODE, new InetSocketAddress(address.address, address.port));
+          Change.REMOVED_NODE, new InetSocketAddress(address.address, address.port), headerFilter);
     }
 
-    public static TopologyChange movedNode(InetAddressAndPort address) {
+    public static TopologyChange movedNode(
+        InetAddressAndPort address, Predicate<Map<String, String>> headerFilter) {
       return new TopologyChange(
-          Change.MOVED_NODE, new InetSocketAddress(address.address, address.port));
+          Change.MOVED_NODE, new InetSocketAddress(address.address, address.port), headerFilter);
     }
 
     // Assumes the type has already been deserialized
     private static TopologyChange deserializeEvent(ByteBuf cb) {
       Change change = CBUtil.readEnumValue(Change.class, cb);
       InetSocketAddress node = CBUtil.readInet(cb);
-      return new TopologyChange(change, node);
+      return new TopologyChange(change, node, c -> true);
```

Comment:
a small thing, we may use:
```
Predicates.alwaysTrue();
```
instead of:
```
 c -> true
```
also, could you elaborate more on which use cases we will pass different type of predicate (other than always true)?